### ISSUE TITLE
enable.xxx scripts in microk8s should be overrideable as well as yaml files!

### ISSUE THREAD

**Issue by typoworx-de** [canonical/microk8s#2548]
While struggling around with ingress (#2546) I figured out some inconsistencies. For most k8s services it is possible to override args or configs. But for ingress this doesn't work.

It should be possible to customize enable.{my-service} and {my-service}.yaml by having a local copy in:
/var/snap/microk8s/current/microk8s-resources/

But this doesn't seem to work.

**Comment by ktsakalozos** [comment]
That is a good suggestion.

We could have the enbale commands search for the `enable.{my-service}` and `{my-service}.yaml` in $SNAP_DATA (that is `/var/snap/microk8s/current/`) and not only in `$SNAP`. This patch should be a relatively easy patch in [1].

[1] https://github.com/ubuntu/microk8s/blob/master/scripts/wrappers/common/utils.py#L249

**Comment by Umesh-01** [comment]
Hello @ktsakalozos, 
I want to work on this if anyone can guide me because I am new to it??

**Comment by ktsakalozos** [comment]
Hi @Umesh-01, that is great to hear.

First, you will need to fork the repository git clone it and try to build it. Have a look at [1].

Next you will need to patch [2] to have the enable script look for addons in a writeable location. 

Come over in the #microk8s channel on the kubernetes.slack.com for real-time discussion!

[1] https://github.com/ubuntu/microk8s/blob/master/docs/build.md
[2] https://github.com/ubuntu/microk8s/blob/master/scripts/wrappers/common/utils.py#L249

**Comment by nikakis** [comment]
Hello @ktsakalozos, as I can see there no changes yet from @Umesh-01. Can I take up this ticket?
Thanks!

**Comment by krisstern** [comment]
@ktsakalozos Been wondering if something simple like the following would suffice?:
```python
if snap_data() / "actions/enable.*.{sh,py}".exists() or snap_data() / "actions/*.yaml".exists():
    actions = snap_data() / "actions"
else:
    actions = Path(__file__).absolute().parent) / "../../../actions"
```
My advance apologies if my attempt seems vain, I am new to the repo and am still learning. Any feedback would be greatly appreciated. Just am really curious to know and learn.

**Comment by neoaggelos** [comment]
This issue is no longer relevant, this has been resolved in MicroK8s 1.24+ with the addition of add-on repositories.
